spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -1,131 +1,505 @@
-            trecent.append(tstart - tcomp[tcomp < tstart].max())
+        return spark_app
 
-        self.stageData["time_since_last_event"] = trecent
+    def compute_all_job_data(self, raw_data: dict, spark_app: SparkApplication) -> SparkApplication:
+        # SPC113 - SDG
+        # Because of the way jobData is created, if there are no job Events in the eventlog then the
+        # correct fields will not exist. A second condition checking for the 'job_id' field is
+        # necessary here to ensure this method will run if this is the case.
+        #
+        # Note: stageData is initialized differently so this same issue does not exist for that
+        # structure. Furthermore, in the event that 'jobData' has no values within, 'stageData' will
+        # also have no values and an invalidLog exception will be thrown during log validation
+        # in SparkApplicationAdvanced.
+        job_data = raw_data.get("jobData", None)
+        if job_data is not None and ("job_id" in job_data):
+            df = pd.DataFrame.from_dict(raw_data["jobData"])
+            df.set_index("job_id")
+            spark_app.jobData = df
+
+        return spark_app
+
+    def compute_all_stage_data(
+        self, raw_data: dict, spark_app: SparkApplication
+    ) -> SparkApplication:
+        if "stageData" in raw_data:
+            spark_app.stageData = pd.DataFrame.from_dict(raw_data["stageData"]).set_index(
+                "stage_id"
+            )
 
-        if self.existsSQL:
-            trecent = []
-            for qid in self.sqlData.index.values:
-                tstart = self.sqlData.loc[qid]["start_time"]
+        return spark_app
 
-                tmp = tcomp[tcomp < tstart]
+    def compute_all_task_data(
+        self, raw_data: dict, spark_app: SparkApplication
+    ) -> SparkApplication:
+        if "taskData" in raw_data:
+            spark_app.taskData = pd.DataFrame.from_dict(raw_data["taskData"]).set_index("task_id")
 
-                if len(tmp) == 0:
-                    trecent.append(0)
-                else:
-                    trecent.append(tstart - tcomp[tcomp < tstart].max())
-            self.sqlData["time_since_last_event"] = trecent
+        return spark_app
 
-    # This method collects all of the sql information into a dataframe
-    def getSQLinfo(self, appobj):
+    def compute_all_driver_accum_data(
+        self, raw_data: dict, spark_app: SparkApplication
+    ) -> SparkApplication:
+        if "accumData" in raw_data:
+            spark_app.accumData = pd.DataFrame.from_dict(raw_data["accumData"])
+            if "sql_id" in spark_app.accumData.columns:
+                spark_app.accumData = spark_app.accumData.set_index("sql_id")
 
+        return spark_app
+
+    def compute_all_metadata(self, raw_data: dict, spark_app: SparkApplication) -> SparkApplication:
+        spark_app.metadata = raw_data.get("metadata", {})
+
+        # TODO - still necessary? Can we delete this?
```

Comment:
Yeah ... it's been a few weeks since March 2022.

Reply:
haha, indeed it has! Just always want to double-check since I didn't have a hand in rolling that out, and sometimes comments lie, haha. Thanks!

Reply:
I'm sure when I wrote that comment originally I said to myself: "I'll come back here in a few weeks to delete this"  lmao

Reply:
tale as old as time, hahaha